regression/scripts/common.sh

```diff
@@ -674,6 +659,77 @@ echo "   ${subproj}: dep_jobids = ${dep_jobids}"
 echo " "
 }
 
+##----------------------------------------------------------------------------##
+# Canonicalize_filename:
+# See http://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac
+##----------------------------------------------------------------------------##
+canonicalize_filename () {
+    local target_file=$1
+    local physical_directory=""
+    local result=""
+
+    # Need to restore the working directory after work.
+    pushd `pwd` > /dev/null
+
+    cd "$(dirname "$target_file")"
+    target_file=`basename $target_file`
+
+    # Iterate down a (possible) chain of symlinks
+    while [ -L "$target_file" ]
+    do
+        target_file=$(readlink "$target_file")
+        cd "$(dirname "$target_file")"
+        target_file=$(basename "$target_file")
+    done
+
+    # Compute the canonicalized name by finding the physical path
+    # for the directory we're in and appending the target file.
+    physical_directory=`pwd -P`
+    result="$physical_directory"/"$target_file"
+
+    # restore the working directory after work.
+    popd > /dev/null
+
+    echo "$result"
+}
+
+##----------------------------------------------------------------------------##
+# Compare versions
+#
+# Example:
+#
+# EMACSVER=`emacs --version | head -n 1 | sed -e 's/.*Emacs //'`
+# if [[ `version_gt "24.0.0" $EMACSVER`  ]]; then echo yes; fi
+##----------------------------------------------------------------------------##
+function version_gt()
+{
+  test "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1";
+}
+
+##----------------------------------------------------------------------------##
+# check whether the given file matches any of the set extensions
+#
+# Example:
+#
+# FILE_EXTS=".f90 .F90"
+# FILE_ENDINGS="_f.h _f77.h _f90.h"
+# for file in $modified_files; do
+#   if ! matches_extension $file; then
+#      continue;
+#   fi
+#   <do stuff>
+# done
+##----------------------------------------------------------------------------##
+matches_extension() {
+    local filename=$(basename "$1")
+    local extension=".${filename##*.}"
+    local end
+    local ext
+    for end in $FILE_ENDINGS; do [[ "$filename" == *"$end" ]] && return 1; done
```

Comment:
This script was moved here from the git-clang-format pre-commit hook sample.  I wanted to leave the function intact as this might have some use in the future (e.g.: `file.f90.in`).